pkg/controller/internal/selector/selector.go

```diff
@@ -1,46 +0,0 @@
-package selector
```

Comment:
It became useful to use the upstream Kubernetes apimachinery labels package over this custom interface so that we could pass a selector from a library consumer to a controller to the collector to the ctrlClient selector without conversions. It makes it a breaking change so I will bump the major version if this removal is acceptable. An example of a PR converting an operator to use this new format can be found here https://github.com/giantswarm/node-operator/pull/181